### PR TITLE
Budget

### PR DESCRIPTION
**Pull request by jonspock**
Superblock rewards with some default Addresses and % rewards,
Currently it occurs every 100 blocks for testing
Let's Squash when we merge, please

### CODE REVIEW

File: src/budget/budget.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#include "amount.h"
+#include "chainparams.h"
+#include "config.h"
+#include "dstencode.h"
+#include "logging.h"
+#include "primitives/transaction.h"
+#include <consensus/validation.h>
+
+// Group Address with % for safety/clarity
+struct BudgetStruct {
```
Comment (joseratts) [inline src/budget/budget.cpp]: Wouldn't be good to have another label field to identify the DAO

File: src/budget/budget.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class CBlock;
+class CMutableTransaction;
+
+class CBudget {
+
+  private:
+    std::unique_ptr<CScript []> Scripts;
+    std::unique_ptr<Amount []> nPayment;
+
+    int64_t nBlocksPerPeriod;
```
Comment (joseratts) [inline src/budget/budget.h]: Should this at some point be equal to 100? for our testing purposes

File: src/budget/budget.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  return fPaymentOK;
+}
+
+Amount CBudget::CalculateSuperBlockRewards(int nHeight, const Amount &nOverallReward) {
+  if (!IsSuperBlock(nHeight)) return Amount();
+
+  // Overall Reward should be an integer
+  Amount sumRewards;
+  for (int i = 0; i < BudgetSize; i++) {
+    nPayment[i] = (((Payouts[i].percent * nBlocksPerPeriod * nOverallReward) / COIN) * COIN) / 100;
```
Comment (joseratts) [inline src/budget/budget.cpp]: Could we get rid of COIN? COIN/COIN = 1

File: src/budget/budget.cpp (hunk 1 above)
Comment (jonspock) [reply src/budget/budget.cpp]: Makes sense

File: src/budget/budget.cpp (hunk 3 above)
Comment (jonspock) [reply src/budget/budget.cpp]: The purpose was to quantize the Payment to whole COIN amounts, need to double check. Also the % is not quite right yet

**Pull request merged**